TransIT.BLL/Services/UserService.cs

```diff
@@ -0,0 +1,128 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using TransIT.BLL.Security.Hashers;
+using TransIT.DAL.Models.Entities;
+using TransIT.DAL.Repositories.InterfacesRepositories;
+using TransIT.DAL.UnitOfWork;
+
+namespace TransIT.BLL.Services
+{
+    /// <summary>
+    /// User model CRUD
+    /// </summary>
+    public class UserService : IUserService
+    {
+        /// <summary>
+        /// Saves changes
+        /// </summary>
+        private readonly IUnitOfWork _uow;
+        /// <summary>
+        /// User repo
+        /// </summary>
+        private readonly IUserRepository _userRepository;
+        /// <summary>
+        /// Roles repo
+        /// </summary>
+        private readonly IRoleRepository _roleRepository;
+        /// <summary>
+        /// Hasher for password
+        /// </summary>
+        private readonly IPasswordHasher _hasher;
+        
+        /// <summary>
+        /// Ctor
+        /// </summary>
+        /// <param name="hasher">Used to hash password</param>
+        /// <param name="uow">Used to save changes</param>
+        /// <param name="userRepository">Used to CRUD users</param>
+        /// <param name="roleRepository">Used to retrieve roles</param>
+        public UserService(
+            IPasswordHasher hasher,
+            IUnitOfWork uow,
+            IUserRepository userRepository,
+            IRoleRepository roleRepository)
+        {
+            _uow = uow;
+            _userRepository = userRepository;
+            _roleRepository = roleRepository;
+            _hasher = hasher;
+        }
+
+        /// <summary>
+        /// Finds user by id
+        /// </summary>
+        /// <param name="userId">Id of user</param>
+        /// <returns>User or null unless not found</returns>
+        public Task<User> Get(int userId) =>
+            _userRepository.GetByIdAsync(userId);
+
+        /// <summary>
+        /// Gives users with pagination
+        /// </summary>
+        /// <param name="index">Where to start</param>
+        /// <param name="amount">Amount to give</param>
+        /// <returns>List of users</returns>
+        public Task<IEnumerable<User>> Get(uint index, uint amount) =>
+            _userRepository.GetRangeAsync(index, amount);
+
+        /// <summary>
+        /// Creates user if login and password not empty and does not exist in DB
+        /// hashes password and set zero to id
+        /// </summary>
+        /// <param name="value">User model</param>
+        /// <returns>Is successful</returns>
+        public async Task<bool> Create(User value)
+        {
+            if (string.IsNullOrEmpty(value.Login)
+                || string.IsNullOrEmpty(value.Password)
```

Comment:
I don't like this check
Lower layer will check this and throw an exception if necessary
I suppose that input values are valid always, but it's up to you to choose